src/services/etherspot.js

```diff
@@ -124,8 +118,27 @@ class EtherspotService {
     this.subscription.unsubscribe();
   }
 
-  getAccount(accountAddress: string): Promise<?EtherspotAccount> {
-    return this.sdk.getAccount({ address: accountAddress }).catch((error) => {
+  getSdkForChain(chain: Chain): ?EtherspotSdk {
+    const network = networkNameFromChain(chain);
+    if (!network) {
+      reportErrorLog('EtherspotService getSdkForChain failed', { chain });
+      return undefined;
+    }
+
+    const sdk = this.instances[network];
+    if (!sdk) {
+      reportErrorLog('EtherspotService getSdkForChain failed', { chain, network });
+      return undefined;
+    }
+
+    return sdk;
+  }
+
+  getAccount(chain: Chain, accountAddress: string): Promise<?EtherspotAccount> {
+    const sdk = this.getSdkForChain(chain);
+    if (!sdk) return Promise.resolve(null);
```

Comment:
It can be just `null` I guess
```suggestion
    if (!sdk) return null;
```

Reply:
Yes, but had to updated return type to `Promise | null`